okapi-core/src/main/java/org/folio/okapi/bean/ModuleDescriptor.java

```diff
@@ -294,8 +295,21 @@ public void setLaunchDescriptor(LaunchDescriptor launchDescriptor) {
     return permissionSets;
   }
 
+  /**
+   * Set permission sets.
+   * @param permissionSets permission sets
+   * @throws IllegalArgumentException if permissionName is missing
+   */
   public void setPermissionSets(Permission[] permissionSets) {
     this.permissionSets = permissionSets;
+    if (permissionSets != null) {
+      for (Permission permission : permissionSets) {
+        String permissionName = permission.getPermissionName();
+        if (permissionName == null) {
+          throw new IllegalArgumentException("Missing permissionName");
+        }
+      }
+    }
```

Comment:
Maybe move these before Line 304? It won't make a difference here, but validation before assign seems more natural.

Reply:
ok